library/azure_rm_devtestlabslab.py

```diff
@@ -0,0 +1,362 @@
+    from msrest.polling import LROPoller
+    from msrestazure.azure_operation import AzureOperationPoller
+    from azure.mgmt.devtestlabs import DevTestLabsClient
+    from msrest.serialization import Model
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class Actions:
+    NoAction, Create, Update, Delete = range(4)
+
+
+class AzureRMLabs(AzureRMModuleBase):
+    """Configuration class for an Azure RM Lab resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            location=dict(
+                type='str'
+            ),
+            lab_storage_type=dict(
+                type='str',
+                choices=['standard',
+                         'premium']
+            ),
+            premium_data_disks=dict(
+                type='bool'
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.lab = dict()
+
+        self.results = dict(changed=False)
+        self.mgmt_client = None
+        self.state = None
+        self.to_do = Actions.NoAction
+
+        super(AzureRMLabs, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                          supports_check_mode=True,
+                                          supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+            elif kwargs[key] is not None:
+                self.lab[key] = kwargs[key]
+
+        expand(self.lab, ['lab_storage_type'], camelize=True)
+        expand(self.lab, ['premium_data_disks'], map={True: 'Enabled', False: 'Disabled'})
+
+        response = None
+
+        self.mgmt_client = self.get_mgmt_svc_client(DevTestLabsClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        resource_group = self.get_resource_group(self.resource_group)
+
+        old_response = self.get_lab()
+
+        if not old_response:
+            self.log("Lab instance doesn't exist")
+            if self.state == 'absent':
+                self.log("Old instance didn't exist")
+            else:
+                self.to_do = Actions.Create
+        else:
+            self.log("Lab instance already exists")
+            if self.state == 'absent':
+                self.to_do = Actions.Delete
+            elif self.state == 'present':
+                if (not default_compare(self.lab, old_response, '')):
```

Comment:
does everything in the lab object updatable?

Reply:
actually this is a very good generic question. what should we do if something can't be updated?
if we ignore such parameters here, module won't be detecting change at all, and the state defined in playbook 
if we detect change in parameters that cannot be applied, user will get failure error from the cloud.

Reply:
just 2 parameters. changed the check now